api/src/paths/dwc/submission/{datasetId}/artifacts.ts

```diff
@@ -0,0 +1,83 @@
+import { RequestHandler } from 'express';
+import { Operation } from 'express-openapi';
+import { getAPIUserDBConnection, getDBConnection } from '../../../../database/db';
+import { defaultErrorResponses } from '../../../../openapi/schemas/http-responses';
+import { ArtifactService } from '../../../../services/artifact-service';
+import { getLogger } from '../../../../utils/logger';
+
+const defaultLog = getLogger('paths/dwc/eml/get');
+
+export const GET: Operation = [getArtifactsByDatasetId()];
+
+GET.apiDoc = {
+  description: 'Retrieves dataset artifact records by dataset ID',
+  tags: ['artifacts'],
+  security: [
+    {
+      OptionalBearer: []
+    }
+  ],
+  parameters: [
+    {
+      description: 'Dataset identifier',
+      in: 'path',
+      name: 'datasetId',
+      schema: {
+        type: 'string'
+      },
+      required: true
+    }
+  ],
+  responses: {
+    200: {
+      description: 'Dataset attachments response object.',
+      content: {
+        'application/json': {
+          schema: {
+            type: 'object',
+            properties: {
+              artifacts: {
+                type: 'array',
+                items: {
+                  type: 'object'
```

Comment:
It would be ideal if we could spec out the response more fully here.

(as a side note, there are some existing efforts to build a zod->swagger converter library, but they are a bit underdeveloped at the moment last I looked, though gaining popularity as zod is blowing up)

Reply:
Good catch, I'll flesh this out (I told myself I would do it later and forgot 🤥)